ipatests/test_integration/test_ipahealthcheck.py

```diff
@@ -1603,3 +1612,361 @@ def test_append_arguments_to_list_sources(self):
+        """
+        Fixture to remove and add ca using getcert command.
+        """
+        self.master.run_command(
+            ["getcert", "remove-ca", "-c", "dogtag-ipa-ca-renew-agent"]
+        )
+        yield
+        self.master.run_command(
+            [
+                "getcert",
+                "add-ca",
+                "-c",
+                "dogtag-ipa-ca-renew-agent",
+                "-e",
+                paths.DOGTAG_IPA_CA_RENEW_AGENT_SUBMIT,
+            ]
+        )
+
+    def test_ipahealthcheck_certmongerca(self, getcert_ca):
+        """
+        Test that healthcheck detects that a certmonger-defined
+        CA is missing
+        """
+        returncode, data = run_healthcheck(
+            self.master, "ipahealthcheck.ipa.certs", "IPACertmongerCA",
+        )
+        assert returncode == 1
+        for check in data:
+            if check["kw"]["key"] == "dogtag-ipa-ca-renew-agent":
+                assert check["result"] == "ERROR"
+                assert (
+                    check["kw"]["msg"]
+                    == "Certmonger CA '{key}' missing"
+                )
+
+    @pytest.fixture()
+    def rename_httpd_cert(self):
+        """
+        Fixture to rename http cert and revert the change.
+        """
+        self.master.run_command(
+            ["mv", paths.HTTPD_CERT_FILE, "%s.old" % paths.HTTPD_CERT_FILE]
+        )
+        yield
+        self.master.run_command(
+            ["mv", "%s.old" % paths.HTTPD_CERT_FILE, paths.HTTPD_CERT_FILE]
+        )
+
+    def test_ipahealthcheck_ipaopensslchainvalidation(self, rename_httpd_cert):
+        """
+        Test for IPAOpenSSLChainValidation when httpd cert is moved.
+        """
+        error_msg = "Can't open {} for reading".format(paths.HTTPD_CERT_FILE)
+        returncode, data = run_healthcheck(
+            self.master,
+            "ipahealthcheck.ipa.certs",
+            "IPAOpenSSLChainValidation",
+        )
+        assert returncode == 1
+        for check in data:
+            if check["kw"]["key"] == paths.HTTPD_CERT_FILE:
+                assert check["result"] == "ERROR"
+                assert error_msg in check["kw"]["reason"]
+
+    @pytest.fixture()
+    def replace_ipa_chain(self):
+        """
+        Fixture to drop the external CA from the IPA chain
+        """
+        self.master.run_command(
+            ["cp", paths.IPA_CA_CRT, "%s.old" % paths.IPA_CA_CRT]
+        )
+        self.master.run_command(
+            [paths.CERTUTIL,
+             "-d", paths.PKI_TOMCAT_ALIAS_DIR,
+             "-L",
+             "-a",
+             "-n", "caSigningCert cert-pki-ca",
+             "-o", paths.IPA_CA_CRT]
+        )
+        yield
+        self.master.run_command(
+            ["mv", "%s.old" % paths.IPA_CA_CRT, paths.IPA_CA_CRT]
+        )
+
+    def test_opensslchainvalidation_ipa_ca_cert(self, replace_ipa_chain):
+        """
+        Test for IPAOpenSSLChainValidation when IPA CA cert is renamed
```

Comment:
The comment needs to be fixed: when /etc/ipa/ca.crt contains IPA CA cert but not the external CA.

Reply:
Comment has been fixed, also the yaml files have been updated to include the ExternalCA test in nightly_* yaml files